File: 1737-maximum-nesting-depth-of-the-parentheses/maximum-nesting-depth-of-the-parentheses.ts
function maxDepth(s: string): number {
    let count = 0;
    let maxNum = 0;
    for (let c of s) {
        if (c === '(') {
            count += 1;
            if (maxNum < count)
                maxNum = count;
        } else if (c === ')') {
            count -= 1;
        }
    }
    return maxNum;
};

File: 1737-maximum-nesting-depth-of-the-parentheses/maximum-nesting-depth-of-the-parentheses.java
class Solution {
    public int maxDepth(String s) {
        int count = 0;
        int maxNum = 0;
        for (char c : s.toCharArray()) {
            if (c == '(') {
                count += 1;
                if (maxNum < count)
                    maxNum = count;
            } else if (c == ')') {
                count -= 1;
            }
        }
        return maxNum;
    }
}